gm4_better_fire/data/better_fire/functions/main.mcfunction

```diff
@@ -1,6 +1,6 @@
 #fire arrows place fire
-execute at @e[nbt=!{Fire:-1s},type=arrow] run setblock ~ ~ ~ fire keep
+execute as @e[nbt=!{Fire:-1s},type=minecraft:arrow] at @s run function better_fire:checks

```

Comment:
Put type before nbt for efficiency and move the check from `better_fire:checks` into this command. Probably rename the checks function
```suggestion
execute as @e[type=minecraft:arrow,nbt=!{Fire:-1s},nbt={inBlockState:{}}] at @s if block ~ ~ ~ #better_fire:air run function better_fire:in_block
```